### PR TITLE
dasharo-compatibility/326-DTS-firmware-update.md: add new test case

### PR DESCRIPTION
**Pull request by Pokisiekk**
Signed-off-by: Kamil Pokornicki <kamil.pokornicki@3mdeb.com>

### CODE REVIEW

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 1]
```diff
@@ -12,7 +12,53 @@
...
+In the `Setup menu` information about the current firmware version should be
+displayed.
+
+Example output:
+
+```bash
+Dasharo (coreboot+UEFI) v1.2.0
+```
+
+### FDT002.001 Firmware update by using Dasharo Tools Suite
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:61]: I think at this point we should still point out that for flashing purposes we use fwupd.

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 2]
```diff
@@ -12,7 +12,53 @@
...
+1. In the `Network Boot menu`, select the `Dasharo Tools Suite` option.
+1. Login as root (no password required).
+1. Type 9 and click `Enter` to launch into Shell.
+1. Run the following commands to update firmware to the latest version:
+
+    ```bash
+    flashrom -p internal -w /tmp/coreboot.rom
+    ```
+
+1. Reboot the DUT.
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:46]: This is not a universal solution.

I think that we should at this point use the command  'dmidecode -t 0` written in the DTS shell.

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 3]
```diff
@@ -12,7 +12,53 @@
...
+1. Press `SETUP_MENU_KEY` to enter the setup menu.
+1. Note the results.
+
+**Expected result**
+
+In the `Setup menu` information about the current firmware version should be
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:52]: As above, at this point we should show output from the above-described command.

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 4]
```diff
@@ -12,7 +12,53 @@
...
+1. Proceed with the [Common](#common) section.
+1. [Disable Secure Boot](../../unified-test-documentation/dasharo-security/206-secure-boot.md).
+1. Obtain `coreboot.rom` binary.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Press `BOOT_MENU_KEY` to enter the boot menu.
+1. In the `Boot Menu`, select the `iPXE Network Boot` option.
+1. In the `Network Boot menu`, select the `Dasharo Tools Suite` option.
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:37]: This is not a universal solution. Some devices have hardcoded Dasharo Tools Suite in the ipxe menu, some do not have such an option, so we should use ipxe shell and need a net interface to be configured.

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 1 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md]: added

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 2 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md]: appiled

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 4 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md]: appiled

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 5]
```diff
@@ -12,7 +12,62 @@
...
+
+1. Power on the DUT.
+1. Press `BOOT_MENU_KEY` to enter the boot menu.
+1. In the `Boot Menu`, select the `iPXE Network Boot` option.
+1. In the `Network Boot menu`, select the `iPXE Shell` option.
+1. Type in `dhcp` to obtain an IP address.
+1. Type in `chain  http://boot.3mdeb.com/dts.ipxe` to load DTS.
+1. Wait for `Enter an option`.
+1. Type 9 and click `Enter` to launch into Shell.
+1. Run the following commands to flash firmware:
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:42]: Missing information: how to get the correct coreboot binary.

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 6]
```diff
@@ -33,26 +88,36 @@ firmware by using Dasharo Tools Suite (DTS).
...
-displayed.
+The output of `dmidecode` command should contain information about current
+firmware. The current firmware version should be equal to the binary version,
+which you were flashing.
 
 Example output:
 
 ```bash
-Dasharo (coreboot+UEFI) v1.2.0
+Version: Dasharo (coreboot+UEFI) v1.1.0
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:122]:
```suggestion
Dasharo (coreboot+UEFI) v1.1.0
```
The situation described here may appear only on older platforms, or those with dedicated sources.

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md [hunk 7]
```diff
@@ -12,7 +12,62 @@
...
+**Expected result**
+
+The output of `dmidecode` command should contain information about current
+firmware. The current firmware version should be equal to the binary version,
+which you were flashing.
+
+Example output:
+
+```bash
+Version: Dasharo (coreboot+UEFI) v1.1.0
```
Comment (sulewskiprzemyslaw) [inline docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:67]:
```suggestion
Dasharo (coreboot+UEFI) v1.1.0
```
The situation described here may appear only on older platforms, or those with dedicated sources.

**Review by sulewskiprzemyslaw — CHANGES_REQUESTED** [review]

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 6 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:122]: The situation above have place always after dmidecode command - 
Version : Dasharo ([major_framework]+[minor_framework]) v[version]

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 7 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:67]: The situation above have place always after dmidecode command - 
Version : Dasharo ([major_framework]+[minor_framework]) v[version]

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 5 above)
Comment (Pokisiekk) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md]: added

File: docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md (hunk 7 above)
Comment (sulewskiprzemyslaw) [reply docs/unified-test-documentation/dasharo-compatibility/326-DTS-firmware-update.md:75]: OK, thanks for information

**Pull request merged**